// @wix/auto_sdk_pricing-plans_plans/build/cjs/src/pricing-plans-v2-plan-plans.http.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.archivePlan = exports.clearPrimary = exports.makePlanPrimary = exports.setPlanVisibility = exports.updatePlan = exports.createPlan = exports.getPlanStats = exports.listPlans = exports.getPlan = exports.queryPublicPlans = exports.listPublicPlans = void 0;
const rest_modules_1 = require("@wix/sdk-runtime/rest-modules");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const timestamp_2 = require("@wix/sdk-runtime/transformations/timestamp");
const field_mask_1 = require("@wix/sdk-runtime/transformations/field-mask");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const rest_modules_2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressMembershipV2PlansPlansServiceV2Url(opts) {
    const domainToMappings = {
        'api._api_base_domain_': [
            {
                srcPath: '/plan-proxy',
                destPath: '',
            },
        ],
        '*.dev.wix-code.com': [
            {
                srcPath: '/_api/paid-plans/v2/plans',
                destPath: '/v2/plans',
            },
        ],
        '*.pub.wix-code.com': [
            {
                srcPath: '/_api/paid-plans/v2/plans',
                destPath: '/v2/plans',
            },
        ],
        '*.wixapps.net': [
            {
                srcPath: '/_api/paid-plans/v2/plans',
                destPath: '/v2/plans',
            },
        ],
        _: [
            {
                srcPath: '/_api/paid-plans/v2/plans',
                destPath: '/v2/plans',
            },
        ],
        'editor-flow.wixapps.net': [
            {
                srcPath: '/_api/paid-plans/v2/plans',
                destPath: '/v2/plans',
            },
        ],
        'editor._base_domain_': [
            {
                srcPath: '/_api/paid-plans/v2/plans',
                destPath: '/v2/plans',
            },
        ],
        'blocks._base_domain_': [
            {
                srcPath: '/_api/paid-plans/v2/plans',
                destPath: '/v2/plans',
            },
        ],
        'create.editorx': [
            {
                srcPath: '/_api/paid-plans/v2/plans',
                destPath: '/v2/plans',
            },
        ],
        'editor.wixapps.net': [
            {
                srcPath: '/_api/paid-plans/v2/plans',
                destPath: '/v2/plans',
            },
        ],
        'forums._base_domain_': [
            {
                srcPath: '/_api/paid-plans/v2/plans',
                destPath: '/v2/plans',
            },
        ],
        'forums.wixapps.net': [
            {
                srcPath: '/_api/paid-plans/v2/plans',
                destPath: '/v2/plans',
            },
        ],
        'manage._base_domain_': [
            {
                srcPath: '/_api/paid-plans/v2/plans',
                destPath: '/v2/plans',
            },
            {
                srcPath: '/_api/pricing-plans/v2/plans',
                destPath: '/v2/plans',
            },
        ],
        'www._base_domain_': [
            {
                srcPath: '/_api/paid-plans/v2/plans',
                destPath: '/v2/plans',
            },
        ],
        'www.wixapis.com': [
            {
                srcPath: '/pricing-plans/v2/plans',
                destPath: '/v2/plans',
            },
        ],
    };
    return (0, rest_modules_2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
const PACKAGE_NAME = '@wix/auto_sdk_pricing-plans_plans';
/** Retrieves a list of up to 100 public pricing plans. Public plans are visible plans that are available to site visitors and can be purchased. */
function listPublicPlans(payload) {
    function __listPublicPlans({ host }) {
        const metadata = {
            entityFqdn: 'wix.pricing_plans.v2.plan',
            method: 'GET',
            methodFqn: 'com.wixpress.membership.v2.plans.PlansServiceV2.ListPublicPlans',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressMembershipV2PlansPlansServiceV2Url({
                protoPath: '/v2/plans/public',
                data: payload,
                host,
            }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'plans.createdDate' },
                        { path: 'plans.updatedDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __listPublicPlans;
}
exports.listPublicPlans = listPublicPlans;
/**
 * Creates a query to retrieve a list of public pricing plans.
 *
 *
 * The `queryPublicPlans()` method builds a query to retrieve a list of up to 1,000 public plans and returns a [`PublicPlansQueryBuilder`](#plansquerybuilder) object.
 *
 * The returned object contains the query definition which is typically used to run the query using the [`find()`](#plansquerybuilder/find) method.
 *
 * You can refine the query by chaining `PublicPlansQueryBuilder` methods onto the query. `PublicPlansQueryBuilder` methods enable you to sort, filter, and control the results that `queryPublicPlans()` returns.
 *
 * `queryPublicPlans()` runs with the following `PublicPlansQueryBuilder` defaults that you can override:
 * - [`skip`](#plansquerybuilder/skip): `0`
 * - [`limit`](#plansquerybuilder/limit): `50`
 *
 * The methods that are chained to `queryPublicPlans()` are applied in the order they are called. For example, if you sort on the `_createdDate` property in ascending order and then on the `_id` property in ascending order, the results are sorted first by the created date of the items and then, if there are multiple results with the same date, the items are sorted by `_id` in ascending order, per created date value.
 *
 * The following `PublicPlansQueryBuilder` methods are supported for the `queryPublicPlans()` method. For a full description of the Plans object, see the object returned for the [`items`](#plansqueryresult/items) property in [`PublicPlansQueryResult`](#plansqueryresult).
 */
function queryPublicPlans(payload) {
    function __queryPublicPlans({ host }) {
        const metadata = {
            entityFqdn: 'wix.pricing_plans.v2.plan',
            method: 'POST',
            methodFqn: 'com.wixpress.membership.v2.plans.PlansServiceV2.QueryPublicPlans',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressMembershipV2PlansPlansServiceV2Url({
                protoPath: '/v2/plans/public/query',
                data: payload,
                host,
            }),
            data: payload,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'plans.createdDate' },
                        { path: 'plans.updatedDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __queryPublicPlans;
}
exports.queryPublicPlans = queryPublicPlans;
/** Retrieves a pricing plan by ID. */
function getPlan(payload) {
    function __getPlan({ host }) {
        const metadata = {
            entityFqdn: 'wix.pricing_plans.v2.plan',
            method: 'GET',
            methodFqn: 'com.wixpress.membership.v2.plans.PlansServiceV2.GetPlan',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressMembershipV2PlansPlansServiceV2Url({
                protoPath: '/v2/plans/{id}',
                data: payload,
                host,
            }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [{ path: 'plan.createdDate' }, { path: 'plan.updatedDate' }],
                },
            ]),
        };
        return metadata;
    }
    return __getPlan;
}
exports.getPlan = getPlan;
/** Retrieves a list of up to 100 pricing plans (including public, hidden, and archived plans). */
function listPlans(payload) {
    function __listPlans({ host }) {
        const metadata = {
            entityFqdn: 'wix.pricing_plans.v2.plan',
            method: 'GET',
            methodFqn: 'com.wixpress.membership.v2.plans.PlansServiceV2.ListPlans',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressMembershipV2PlansPlansServiceV2Url({
                protoPath: '/v2/plans',
                data: payload,
                host,
            }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'plans.createdDate' },
                        { path: 'plans.updatedDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __listPlans;
}
exports.listPlans = listPlans;
/** Gets statistics about the pricing plans. Currently provides only the total number of pricing plans, including archived plans. */
function getPlanStats(payload) {
    function __getPlanStats({ host }) {
        const metadata = {
            entityFqdn: 'wix.pricing_plans.v2.plan',
            method: 'GET',
            methodFqn: 'com.wixpress.membership.v2.plans.PlansServiceV2.GetPlanStats',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressMembershipV2PlansPlansServiceV2Url({
                protoPath: '/v2/plans/stats',
                data: payload,
                host,
            }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
        };
        return metadata;
    }
    return __getPlanStats;
}
exports.getPlanStats = getPlanStats;
/**
 * Creates a pricing plan. The specified `plan` object must contain a pricing model. A pricing model can be 1 of the following:
 * - **A subscription**: A subscription with recurring payments and how often the plan occurs. Subscriptions can have free trial days.
 * - **A plan that doesn't renew**: A single payment for a specific duration that doesn't renew.
 * - **An unlimited plan**: A single payment for an unlimited amount of time (until canceled).
 * Pricing plans are available to the Wix user in the Pricing Plans section in a site's dashboard.
 */
function createPlan(payload) {
    function __createPlan({ host }) {
        const serializedData = (0, transform_paths_1.transformPaths)(payload, [
            {
                transformFn: timestamp_1.transformSDKTimestampToRESTTimestamp,
                paths: [{ path: 'plan.createdDate' }, { path: 'plan.updatedDate' }],
            },
        ]);
        const metadata = {
            entityFqdn: 'wix.pricing_plans.v2.plan',
            method: 'POST',
            methodFqn: 'com.wixpress.membership.v2.plans.PlansServiceV2.CreatePlan',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressMembershipV2PlansPlansServiceV2Url({
                protoPath: '/v2/plans',
                data: serializedData,
                host,
            }),
            data: serializedData,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [{ path: 'plan.createdDate' }, { path: 'plan.updatedDate' }],
                },
            ]),
        };
        return metadata;
    }
    return __createPlan;
}
exports.createPlan = createPlan;
/**
 * Updates a pricing plan.
 * Updating a plan doesn't impact existing orders made for the plan. All orders keep the details of the original plan that was active at the time of purchase.
 */
function updatePlan(payload) {
    function __updatePlan({ host }) {
        const serializedData = (0, transform_paths_1.transformPaths)(payload, [
            {
                transformFn: field_mask_1.transformSDKFieldMaskToRESTFieldMask,
                paths: [{ path: 'fieldMask' }],
            },
            {
                transformFn: timestamp_1.transformSDKTimestampToRESTTimestamp,
                paths: [{ path: 'plan.createdDate' }, { path: 'plan.updatedDate' }],
            },
        ]);
        const metadata = {
            entityFqdn: 'wix.pricing_plans.v2.plan',
            method: 'PATCH',
            methodFqn: 'com.wixpress.membership.v2.plans.PlansServiceV2.UpdatePlan',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressMembershipV2PlansPlansServiceV2Url({
                protoPath: '/v2/plans/{plan.id}',
                data: serializedData,
                host,
            }),
            data: serializedData,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [{ path: 'plan.createdDate' }, { path: 'plan.updatedDate' }],
                },
            ]),
        };
        return metadata;
    }
    return __updatePlan;
}
exports.updatePlan = updatePlan;
/**
 * Sets visibility for pricing plans. Visible plans are considered public plans.
 * By default, pricing plans are public, meaning they are visible. Plans can be hidden so that site members and visitors can't choose them.
 * As opposed to archiving, setting visibility can be reversed. This means that a public plan can be hidden, and a hidden plan can be made public (visible). (An archived plan always remains archived and can't be made active again.)
 * Changing a plan’s visibility doesn't impact existing orders for the plan. All orders for hidden plans are still active and keep their perks.
 */
function setPlanVisibility(payload) {
    function __setPlanVisibility({ host }) {
        const metadata = {
            entityFqdn: 'wix.pricing_plans.v2.plan',
            method: 'PUT',
            methodFqn: 'com.wixpress.membership.v2.plans.PlansServiceV2.SetPlanVisibility',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressMembershipV2PlansPlansServiceV2Url({
                protoPath: '/v2/plans/{id}/visibility',
                data: payload,
                host,
            }),
            data: payload,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [{ path: 'plan.createdDate' }, { path: 'plan.updatedDate' }],
                },
            ]),
        };
        return metadata;
    }
    return __setPlanVisibility;
}
exports.setPlanVisibility = setPlanVisibility;
/** Marks a pricing plan as the primary pricing plan. When viewing pricing plans on a site, the primary plan is highlighted with a customizable ribbon. Only a single plan can be marked as a primary plan at any given time. If there is an existing plan marked as primary, calling Make Plan Primary causes the existing primary plan to lose its primary status. */
function makePlanPrimary(payload) {
    function __makePlanPrimary({ host }) {
        const metadata = {
            entityFqdn: 'wix.pricing_plans.v2.plan',
            method: 'POST',
            methodFqn: 'com.wixpress.membership.v2.plans.PlansServiceV2.MakePlanPrimary',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressMembershipV2PlansPlansServiceV2Url({
                protoPath: '/v2/plans/{id}/make-primary',
                data: payload,
                host,
            }),
            data: payload,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [{ path: 'plan.createdDate' }, { path: 'plan.updatedDate' }],
                },
            ]),
        };
        return metadata;
    }
    return __makePlanPrimary;
}
exports.makePlanPrimary = makePlanPrimary;
/** Sets all pricing plans as not primary. When viewing pricing plans on a site, no plan is highlighted with a customizable ribbon. */
function clearPrimary(payload) {
    function __clearPrimary({ host }) {
        const metadata = {
            entityFqdn: 'wix.pricing_plans.v2.plan',
            method: 'POST',
            methodFqn: 'com.wixpress.membership.v2.plans.PlansServiceV2.ClearPrimary',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressMembershipV2PlansPlansServiceV2Url({
                protoPath: '/v2/plans/clear-primary',
                data: payload,
                host,
            }),
            data: payload,
        };
        return metadata;
    }
    return __clearPrimary;
}
exports.clearPrimary = clearPrimary;
/**
 * Archives a single plan.
 * When a plan is archived, it's no longer visible as a public plan that can be chosen by site members or visitors.
 * This is because the plan's `public` property is automatically set to `false`.
 * Archived plans can't be purchased or reactivated.
 * Plan archiving doesn't impact existing orders made for the plan. All orders for the plan are still active and keep their perks, payment options, and terms.
 * Wix users can see archived plans in a site's dashboard under **Pricing Plans -> Archived Plans**.
 * > **Note:** An attempt to archive an already-archived plan throws an error.
 */
function archivePlan(payload) {
    function __archivePlan({ host }) {
        const metadata = {
            entityFqdn: 'wix.pricing_plans.v2.plan',
            method: 'POST',
            methodFqn: 'com.wixpress.membership.v2.plans.PlansServiceV2.ArchivePlan',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressMembershipV2PlansPlansServiceV2Url({
                protoPath: '/v2/plans/{id}/archive',
                data: payload,
                host,
            }),
            data: payload,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [{ path: 'plan.createdDate' }, { path: 'plan.updatedDate' }],
                },
            ]),
        };
        return metadata;
    }
    return __archivePlan;
}
exports.archivePlan = archivePlan;
